src/ossia/network/base/node_functions.cpp

```diff
@@ -640,6 +642,42 @@ std::vector<ossia::net::node_base*> list_all_children(ossia::net::node_base* nod
   return list;
 }
 
+/**
+ * @brief fuzzysearch: search for nodes that match the pattern string
+ * @param node: root node from where to start
+ * @param pattern: strings to search
+ * @return a vector of pairs of matching score and node pointer sorted in descending order
+ */
+std::vector<std::tuple<double, std::string, ossia::net::node_base*>>
+fuzzysearch(ossia::net::node_base* node, const std::vector<std::string>& patterns)
+{
+  auto children = list_all_children(node);
+
+  std::vector<std::string> oscnames;
+  oscnames.reserve(children.size());
+
+  using score = std::tuple<double, std::string, ossia::net::node_base*>;
+
+  std::vector<score> scores;
+
+  for(const auto& n : children)
```

Comment:
Wonder how feasible it'd be to #pragma omp parallel for ... if this becomes a performance question one day

Reply:
done with fb2a976d